tla-assignments/src/main/scala/at/forsyte/apalache/tla/assignments/Reordering.scala

```diff
@@ -0,0 +1,50 @@
+package at.forsyte.apalache.tla.assignments
+
+import at.forsyte.apalache.tla.lir.oper.TlaBoolOper
+import at.forsyte.apalache.tla.lir.{LetInEx, OperEx, TlaEx, TlaOperDecl}
+import at.forsyte.apalache.tla.lir.transformations.{TlaExTransformation, TransformationTracker}
+
+class Reordering[T]( ord : Ordering[T], rankingFn : TlaEx => T, tracker : TransformationTracker ) {
```

Comment:
I figured this class was tiny enough, but sure, can do.

Reply:
Yeah, the class is tiny enough that it's clear what its doing. But for a reader without the full context (which most will be) there's no clear indication of *why* this would need to be done, imo. Just a line or two indicating why we need a reordering class should be enough of a pointer for future readers I think.